src/Modules/UngradedModules.jl

```diff
@@ -3877,17 +3878,206 @@ function free_resolution(F::FreeMod)
   return res
 end
 
+#= Fill functions (and helpers) for Hecke ChainComplexes in terms of free resolutions =#
+function get_last_map_key(cc::Hecke.ChainComplex)
+  ctr = cc.start
+  while haskey(cc.maps, ctr)
+    ctr -= 1
+  end
+  return ctr+1
+end
+
+function extend_free_resolution(cc::Hecke.ChainComplex, idx::Int; algorithm::String="fres")
+  key = get_last_map_key(cc)
+  if cc.complete == true
+    return cc.maps[key]
+  end
+
+  kernel_entry          = image(cc.maps[key])[1]
+  br                    = base_ring(kernel_entry)
+  singular_free_module  = singular_module(ambient_free_module(kernel_entry))
+  singular_kernel_entry = Singular.Module(base_ring(singular_free_module),
+                              [singular_free_module(repres(g)) for g in gens(kernel_entry)]...)
+  singular_kernel_entry.isGB = true
+
+  len = idx + key - cc.start + 1
+  if algorithm == "fres"
+    res = Singular.fres(singular_kernel_entry, len, "complete")
+  elseif algorithm == "sres"
+    res = Singular.fres(singular_kernel_entry, len)
+  elseif algorithm == "lres"
+    error("LaScala's method is not yet available in Oscar.")
+  end
+
+  if Singular.length(res) < len
+    cc.complete = true
+  end
+  dom = domain(cc.maps[key])
+  j   = 2
+  while j <= Singular.length(res)
+    key -= 1
+    codom = dom
+    dom   = free_module(br, Singular.ngens(res[j]))
+    SM    = SubModuleOfFreeModule(codom, res[j])
+    generator_matrix(SM)
+    map = hom(dom, codom, SM.matrix)
+    cc.maps[key] = map
+    j += 1
+  end
+  # Finalize maps.
+  if cc.complete == true
+    key -= 1
+    Z = FreeMod(br, 0)
+    set_attribute!(Z, :name => "Zero")
+    cc.maps[key] = hom(Z, domain(cc.maps[key+1]), FreeModElem[])
+  end
+  return cc.maps[key]
+end
+
+@doc Markdown.doc"""
+    free_resolution(M::SubQuo; ordering::ModuleOrdering = default_ordering(M),
+        length::Int=0, method::String="complete", algorithm::String="fres")
+
+Return a free resolution of `M`.
+
+If `length != 0`, the free resolution is only computed up to the `length`-th free module.
+`algorithm` can be set to `sres` or `fres`.
+
+# Examples
+```jldoctest
+julia> R, (x, y, z) = PolynomialRing(QQ, ["x", "y", "z"])
+(Multivariate Polynomial Ring in x, y, z over Rational Field, fmpq_mpoly[x, y, z])
+
+julia> A = R[x; y]
+[x]
+[y]
+
+julia> B = R[x^2; x*y; y^2; z^4]
+[x^2]
+[x*y]
+[y^2]
+[z^4]
+
+julia> M = SubQuo(A, B)
+Subquotient of Submodule with 2 generators
+1 -> x*e[1]
+2 -> y*e[1]
+by Submodule with 4 generators
+1 -> x^2*e[1]
+2 -> x*y*e[1]
+3 -> y^2*e[1]
+4 -> z^4*e[1]
+
+julia> fr = free_resolution(M, length=1)
+fr_1 ----> fr_0 ----> M
+where:
+        fr_1 = Free module of rank 6 over Multivariate Polynomial Ring in x, y, z over Rational Field
+        fr_0 = Free module of rank 2 over Multivariate Polynomial Ring in x, y, z over Rational Field
+
```

Comment:
CI fails because these doctests do not match what actually is printed. This is because the "hack" used by Oscar to detect that the computed result was assigned to a variable `fr` does not work inside doctests, so it falls back to generic names.

(Personally, I think we should remove this "guess the names in the REPL" hack, it can be cool when it works but incredibly frustrating in other situations, such as this)

Reply:
This is something we should discuss with @wdecker. I am fine to have this without the "hack", but there may be others insisting on the name guessing for an easier presentation of the resolution. It is not only used for `fr` but also for the start/input `M`  and the end `zero`.

Reply:
Sure, for now the "hack" will stay in any case. 

But regardless of that, this PR won't be merged as long as the doctests fail. So this example needs to be adjusted to use `C_0` instead of `fr_0` etc.

A way out might be to allow users to (optionally) explicitly specify a name (as a symbol argument to `free_resolution`) to use when printing the underlying chain complex, so that it prints e.g. `fr_3` instead of `C_3`. If given, this value would be used; if not given, the "magic"/"hack" heuristic kicks in. But for manual examples, it would at least ensure all looks the way you want, and looks the same on the REPL and in the manual.

Reply:
I thought a bit more about how we could perhaps make the `set_name!` / `find_name` "trick" work in doctests, and here's an idea how we could resolve this (at least for select doctests):
- we add [block-level setup code](https://juliadocs.github.io/Documenter.jl/stable/man/doctests/#Block-level-setup-code) to the affect `jldoctest` which does something akin to `Main.eval(:(ACTIVE_DOCTEST_MODULE = @__MODULE__))`
- then we teach `find_name` to look in `ACTIVE_DOCTEST_MODULE`, if it is defined
- at the end of the doctests, we should reset `ACTIVE_DOCTEST_MODULE`

Of course this may not work quite like that, but I hope the general idea is clear...